### PR TITLE
[test] add script to test which diag commands RCP supports

### PR DESCRIPTION
**Pull request by zhanglongxia**
This commit implements a test `script tools/cp-caps/rcp_caps_test.py` to test which diag commands RCP supports using the DUT and reference device.

### CODE REVIEW

**Review by abtink — APPROVED** [review]
👍

File: tools/cp-caps/README.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+Before running the test script on PC, testers should install dependences first.
+
+```bash
+$ cd tools/otci
+$ pip install build
+$ python -m build
+$ pip install dist/otci_openthread-0.0.1-py3-none-any.whl
+$ pip install adb-shell
+$ pip install adb-shell[usb]
```
Comment (superwhd) [inline tools/cp-caps/README.md:33]: nit: probably you can make a `requirements.txt` so the user don't have to manually install them one by one

File: tools/cp-caps/rcp_caps_test.py [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+        commands = ['diag radio receive', 'diag radio sleep', 'diag radio state']
+
+        if self.__support_commands(commands):
+            self.__dut.diag_radio_receive()
+            receive_state = self.__dut.diag_get_radio_state()
+            self.__dut.wait(0.1)
+            self.__dut.diag_radio_sleep()
+            sleep_state = self.__dut.diag_get_radio_state()
+
+            ret = True if sleep_state == 'sleep' and receive_state == 'receive' else False
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:121]: Will the following work?
```suggestion
            ret = sleep_state == 'sleep' and receive_state == 'receive'
```

File: tools/cp-caps/rcp_caps_test.py [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+        self.__output_results(commands, ret)
+
+    def __test_diag_power(self):
+        power = self.__get_dut_diag_power()
+        commands = ['diag power', f'diag power {power}']
+
+        if self.__support_commands(commands):
+            self.__dut.diag_set_power(power)
+            value = self.__dut.diag_get_power()
+            ret = True if value == power else False
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:105]:
```suggestion
            ret = value == power
```

File: tools/cp-caps/rcp_caps_test.py [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+    # Private methods
+    #
+    def __test_diag_channel(self):
+        channel = 20
+        commands = ['diag channel', f'diag channel {channel}']
+
+        if self.__support_commands(commands):
+            self.__dut.diag_set_channel(channel)
+            channel = self.__dut.diag_get_channel()
+            ret = True if channel == channel else False
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:92]: I don't understand this check. Is here a bug?

File: tools/cp-caps/rcp_caps_test.py [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+        gpio = self.__get_dut_diag_gpio()
+        commands = [f'diag gpio get {gpio}', f'diag gpio set {gpio} 0', f'diag gpio set {gpio} 1']
+
+        if self.__support_commands(commands):
+            self.__dut.diag_set_gpio_value(gpio, 0)
+            value_0 = self.__dut.diag_get_gpio_value(gpio)
+            self.__dut.diag_set_gpio_value(gpio, 1)
+            value_1 = self.__dut.diag_get_gpio_value(gpio)
+
+            ret = True if value_0 == 0 and value_1 == 1 else False
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:137]:
```suggestion
            ret = value_0 == 0 and value_1 == 1
```

File: tools/cp-caps/rcp_caps_test.py [hunk 6]
```diff
@@ -0,0 +1,414 @@
...
+        gpio = self.__get_dut_diag_gpio()
+        commands = [f'diag gpio mode {gpio}', f'diag gpio mode {gpio} in', f'diag gpio mode {gpio} out']
+
+        if self.__support_commands(commands):
+            self.__dut.diag_set_gpio_mode(gpio, 'in')
+            mode_in = self.__dut.diag_get_gpio_mode(gpio)
+            self.__dut.diag_set_gpio_value(gpio, 'out')
+            mode_out = self.__dut.diag_get_gpio_mode(gpio)
+
+            ret = True if mode_in == 'in' and mode_out == 'out' else False
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:153]:
```suggestion
            ret = mode_in == 'in' and mode_out == 'out'
```

File: tools/cp-caps/rcp_caps_test.py [hunk 7]
```diff
@@ -0,0 +1,414 @@
...
+        self.__output_results(commands, ret)
+
+    def __test_diag_echo(self):
+        echo_msg = '0123456789'
+        cmd_diag_echo = f'diag echo {echo_msg}'
+        cmd_diag_echo_num = f'diag echo -n 10'
+
+        if self.__dut.is_command_supported(cmd_diag_echo):
+            reply = self.__dut.diag_echo(echo_msg)
+            ret = True if reply == echo_msg else False
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:166]:
```suggestion
            ret = reply == echo_msg
```

File: tools/cp-caps/rcp_caps_test.py [hunk 8]
```diff
@@ -0,0 +1,414 @@
...
+        else:
+            ret = False
+        self.__output_format_bool(cmd_diag_echo, ret)
+
+        if self.__dut.is_command_supported(cmd_diag_echo_num):
+            reply = self.__dut.diag_echo_number(10)
+            ret = True if reply == echo_msg else False
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:173]:
```suggestion
            ret = reply == echo_msg
```

File: tools/cp-caps/rcp_caps_test.py [hunk 9]
```diff
@@ -0,0 +1,414 @@
...
+                break
+
+        return ret
+
+    def __output_results(self, commands: List[str], support: bool):
+        for command in commands:
+            self.__output_format_bool(command, support)
+
+    def __get_dut_diag_power(self) -> int:
+        return int(os.getenv('DUT_DIAG_POWER')) if os.getenv('DUT_DIAG_POWER') else 10
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:318]: I think you can simplify the code by specifying the `default` parameter for `os.getenv`.

```
os.getenv('DUT_DIAG_POWER', '0')
```

File: tools/cp-caps/rcp_caps_test.py [hunk 10]
```diff
@@ -0,0 +1,414 @@
...
+
+    def __connect_dut(self) -> OTCI:
+        if os.getenv('DUT_ADB_TCP'):
+            node = otci.connect_otbr_adb_tcp(os.getenv('DUT_ADB_TCP'))
+        elif os.getenv('DUT_ADB_USB'):
+            node = otci.connect_otbr_adb_usb(os.getenv('DUT_ADB_USB'))
+        elif os.getenv('DUT_SSH'):
+            node = otci.connect_otbr_ssh(os.getenv('DUT_SSH'))
+        else:
+            self.__fail("Please set DUT_SSH, DUT_ADB_TCP or DUT_SSH to connect to the DUT device.")
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:334]:
```suggestion
            self.__fail("Please set DUT_ADB_TCP, DUT_ADB_USB or DUT_SSH to connect to the DUT device.")
```

File: tools/cp-caps/README.md (hunk 1 above)
Comment (zhanglongxia) [reply tools/cp-caps/README.md]: I have added the `requirements.txt`.

File: tools/cp-caps/rcp_caps_test.py (hunk 4 above)
Comment (zhanglongxia) [reply tools/cp-caps/rcp_caps_test.py]: I have fixed this bug

**Review by superwhd — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**